main.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"crypto/rsa"
+	"github.com/joho/godotenv"
+	"github.com/ufcg-lsd/arrebol-pb-worker/utils"
+	"github.com/ufcg-lsd/arrebol-pb-worker/worker"
+	"log"
+	"os"
+)
+
+const (
+	ConfFilePathKey = "CONF_FILE"
+	ServerEndpointKey = "SERVER_ENDPOINT"
+)
+
+func setup(serverEndPoint string, workerId string) {
+	// The setup routine is responsible for generates the rsa key pairs
+	// of the worker and to send the public part to the server.
+	log.Println("Starting to gen rsa key pair with workerid: " + workerId)
+
+	utils.GenAccessKeys(workerId)
+
+	log.Println("Sending pub key to the server")
+	url := serverEndPoint + "/workers/publicKey"
+	requestBody := &map[string]*rsa.PublicKey{"key": utils.GetPublicKey(workerId)}
+	httpResp := utils.Post(requestBody, url)
+
+	if httpResp.StatusCode != 201 {
+		log.Fatal("Unable to send public key to the server")
+	}
+}
+
+func isTokenValid(token string) bool {
+	return true
+}
+
+func main() {
+	// this main function start the worker following the chosen implementation
+	// passed by arg in the cli. The defaultWorker is started if no arg has been received.
+	err := godotenv.Load()
+
+	if err != nil {
+		log.Println("No .env file found")
+	}
+
+	switch len(os.Args) {
+	case 2:
+		workerImpl := os.Args[1]
+		println(workerImpl)
+	default:
+		defaultWorker()
+	}
+}
+
+func defaultWorker() {
```

Comment:
why are calling this "defaultWorker"? are we planning to have more than one worker type?

Reply:
yeah, maybe this function should be named as defaultWorkerPolicyExecution. As we discussed, the way the worker will execute the tasks can change, it can execute one at a time, or two, or n... For now we only have the trivial one, that stands for one task at a time, and it will be fully implemented by this function...

Reply:
I personally avoid adding abstractions when I do not foresee more than one implementation for it. I would rename this function to avoid the `default`

Reply:
I've renamed it to startWorker()